packages/frontend/src/components/create-campaign/form/restrictions-step/index.tsx

```diff
@@ -0,0 +1,325 @@
+import { Step } from "@/src/components/step";
+import { StepContent } from "@/src/components/step/content";
+import { StepPreview } from "@/src/components/step/preview";
+import {
+    RestrictionType,
+    type CampaignPayload,
+    type CampaignPayloadErrors,
+    type CampaignPayloadPart,
+} from "@/src/types";
+import {
+    Button,
+    ErrorText,
+    Switch,
+    Tabs,
+    Tab,
+    TextInput,
+    Typography,
+} from "@metrom-xyz/ui";
+import { useTranslations } from "next-intl";
+import {
+    useCallback,
+    useEffect,
+    useMemo,
+    useState,
+    type ChangeEvent,
+} from "react";
+import classNames from "classnames";
+import { usePrevious } from "react-use";
+import { isAddress, zeroAddress, type Address } from "viem";
+import { blo } from "blo";
+import { shortenAddress } from "@/src/utils/address";
+import { XIcon } from "@/src/assets/x-icon";
+import { Dot } from "./dot";
+import { CsvAddressesImport } from "./csv-addresses-import";
+
+import styles from "./styles.module.css";
+
+export const MAXIMUM_RESTRICTIONS = 20;
+
+interface RestrictionsStepProps {
+    disabled?: boolean;
+    restrictions?: CampaignPayload["restrictions"];
+    onRestrictionsChange: (restrictions: CampaignPayloadPart) => void;
+    onError: (errors: CampaignPayloadErrors) => void;
+}
+
+export function RestrictionsStep({
+    disabled,
+    restrictions,
+    onRestrictionsChange,
+    onError,
+}: RestrictionsStepProps) {
+    const t = useTranslations("newCampaign.form.restrictions");
+    const [open, setOpen] = useState(false);
+    const [enabled, setEnabled] = useState(false);
+    const [error, setError] = useState("");
+    const [warning, setWarning] = useState("");
+    const [type, setType] = useState(
+        restrictions?.type || RestrictionType.blacklist,
+    );
+    const [address, setAddress] = useState("");
+    const [addresses, setAddresses] = useState<Address[]>(
+        restrictions?.list || [],
+    );
+
+    const prevRestrictions = usePrevious(restrictions);
+
+    const unsavedChanges = useMemo(() => {
+        if (!prevRestrictions && addresses.length === 0) return false;
+        if (!prevRestrictions) return true;
+
+        return (
+            prevRestrictions.list.length !== addresses.length ||
+            prevRestrictions.type !== type
+        );
+    }, [addresses, prevRestrictions, type]);
+
+    // this hooks is used to disable and close the step when
+    // the kpi specification gets disabled, after the campaign creation
```

Comment:
Is this comment still applicable? Or copy/paste issue?

Reply:
Oh, yeah this is still relevant, just fixed the comment 76948d6854810025077649b653ec1f19c908b1ba.